parser-typechecker/src/Unison/ABT.hs

```diff
@@ -224,19 +224,39 @@ freshNamed' used n = fresh' used (v' n)
 -- | `subst v e body` substitutes `e` for `v` in `body`, avoiding capture by
 -- renaming abstractions in `body`
 subst :: (Foldable f, Functor f, Var v) => v -> Term f v a -> Term f v a -> Term f v a
-subst v r t2@(Term fvs ann body)
+subst v r t2 = subst' (const r) v (freeVars r) t2
+
+-- Slightly generalized version of `subst`, the replacement action is handled
+-- by the function `replace`, which is given the annotation `a` at the point
+-- of replacement. `r` should be the set of free variables contained in the
+-- term returned by `replace`. See `substInheritAnnotation` for an example usage.
+subst' :: (Foldable f, Functor f, Var v) => (a -> Term f v a) -> v -> Set v -> Term f v a -> Term f v a
```

Comment:
Is there a coherency problem here? The `Set v` should be computed from the `a`, but that's not enforced. When would the `Set v` be different from the `freeVars` of whatever gets passed to `replace`?

Reply:
Should we choose a naming convention for pseudo-private functions?  A convention other than "has an ugly name".  I bet we have more instances of pseudo-private functions kicking around.